tests/test_ijit/test_ijit_without_numba.py

```diff
@@ -19,5 +19,25 @@ def _simple_func(a, b):
         assert len(w) >= 1
         assert isinstance(_simple_func, types.FunctionType)
 
+@mock.patch.dict(sys.modules, {"numba": None})
+def test_function_given():
+    with warnings.catch_warnings(record=True) as w:
+        custom_jit = reload(einsteinpy.ijit)
+
+        def _simple_func(a, b):
+            return a + b
 
-custom_jit = reload(einsteinpy.ijit)
+        return_func = custom_jit.jit(_simple_func)
```

Comment:
Nice, but you'll miss L17 in `ijit.py`, I think. You need to go one level deeper. Right now, you are calling `ijit()`, but the function that `ijit()` returns is never being called (reason why L17 isn't being covered). Try this:
```
res = custom_jit.jit()

assert res.__name__ == "_jit"
assert res(_simple_func).__name__ == "_simple_func"
```
Also, remove `assert len(w) >= 1`, of course.

Reply:
Rename `test_function_given` to `test_decorator()` and call the other one `test_warning_and_returntype()`.